pkg/gcp/plugin_utils.go

```diff
@@ -290,16 +290,24 @@ func getRegionFromZone(zone string) string {
 	return zone[:strings.LastIndex(zone, "-")]
 }
 
-// Parses GCP compute URL for desired fields
-func parseGCPURL(url string) map[string]string {
-	path := strings.TrimPrefix(url, computeURLPrefix)
-	path = strings.TrimPrefix(path, containerURLPrefix)
-	parsedURL := map[string]string{}
+// parseUrl parses fully qualified and partial GCP resource URLs into a map of URL components (e.g., region)
+// and their corresponding values (e.g., us-west1).
+func parseUrl(url string) map[string]string {
+	path := strings.TrimPrefix(url, computeUrlPrefix)
```

Comment:
Nit: add we add more resources, might be good to have a list of supported ones and iterate through it instead of having to add to this every time we add one